### PR TITLE
mediaSessionAPI: validate metadata

### PR DESCRIPTION
**Pull request by aghassemi**
not sure what browsers do with invalid/dangerous src, but best not to rely on it.

### CODE REVIEW

File: test/functional/test-mediasession-helper.js [hunk 1]
```diff
@@ -135,4 +135,38 @@ describes.sandboxed('MediaSessionAPI Helper Functions', {}, () => {
     const newMetaData = ampdoc.win.navigator.mediaSession.metadata;
     expect(newMetaData).to.deep.equal(fakeMetaData);
   });
+
+  it('should throw is artwork src is invalid - array - object', () => {
```
Comment (wassgha) [inline test/functional/test-mediasession-helper.js]: nit: should throw *if*

File: test/functional/test-mediasession-helper.js [hunk 2]
```diff
@@ -135,4 +135,38 @@ describes.sandboxed('MediaSessionAPI Helper Functions', {}, () => {
...
+      'album': '',
+      'artwork': [
+        {'src': 'javascript://alert(1)'},
+      ],
+      'title': '',
+    };
+    expect(() => {setMediaSession(ampdoc.win, fakeMetaData);}).to.throw();
+  });
+
+  it('should throw is artwork src is invalid - array - string', () => {
```
Comment (wassgha) [inline test/functional/test-mediasession-helper.js]: ditto

File: src/mediasession-helper.js [hunk 3]
```diff
@@ -128,3 +133,22 @@ export function parseFavicon(doc) {
...
+  if (metadata && metadata.artwork) {
+    if (isArray(metadata.artwork)) {
+      metadata.artwork.forEach(artwork => {
+        if (artwork) {
+          const src = isObject(artwork) ? artwork.src : artwork;
+          user().assert(isProtocolValid(src));
+        }
+      });
+    } else {
+      user().assert(isProtocolValid(metadata.artwork));
```
Comment (wassgha) [inline src/mediasession-helper.js]: Throw here, artwork can't be just a `string`, has to be an array of MediaImages [https://wicg.github.io/mediasession/#mediametadata](https://wicg.github.io/mediasession/#mediametadata)

File: src/mediasession-helper.js [hunk 4]
```diff
@@ -128,3 +133,22 @@ export function parseFavicon(doc) {
...
+/**
+ * @private
+ */
+function validateMetadata(metadata) {
+  // Ensure src of artwork has valid protocol
+  if (metadata && metadata.artwork) {
+    if (isArray(metadata.artwork)) {
+      metadata.artwork.forEach(artwork => {
+        if (artwork) {
+          const src = isObject(artwork) ? artwork.src : artwork;
```
Comment (wassgha) [inline src/mediasession-helper.js]: ditto

**Comment by aghassemi** [comment]
done, PTAL.

File: test/functional/test-mediasession-helper.js [hunk 5]
```diff
@@ -141,6 +141,7 @@ describes.sandboxed('MediaSessionAPI Helper Functions', {}, () => {
       'artist': '',
       'album': '',
       'artwork': [
+        /*eslint no-script-url: 0*/
```
Comment (wassgha) [inline test/functional/test-mediasession-helper.js]: I think this disables for whole file so no need for second `eslint no-script-url` right?

**Pull request merged**